reconcile/utils/terraform_client.py

```diff
@@ -575,34 +577,74 @@ def cleanup(self):
         for _, wd in self.working_dirs.items():
             shutil.rmtree(wd)
 
-    def _is_ignored_rds_modification(
+    def _get_pending_rds_modification(
         self, account_name: str, resource_name: str, resource_change: Mapping[str, Any]
-    ) -> bool:
+    ) -> list[str]:
         """
-        Determine whether the RDS resource changes are cases where a terraform apply
+        Determine whether the RDS resource changes are cases where a Terraform apply
         should be skipped.
         """
+        # A commonly changed RDS database settings that can be applied either
+        # immediately or during the next scheduled maintenance window, see:
+        #  https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/Overview.DBInstance.Modifying.html#USER_ModifyInstance.Settings
+        #
+        # Note that changing the RDS database identifier (the database name) is
+        # a complex operation that also involves Terraform performing a resource
+        # replacement and thus it's not included here, see:
+        #   https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/USER_RenameInstance.html
+        supported_values: dict[str, str] = {
```

Comment:
Did you pick the subset of operations? And by which means? I found more operations, where applying immediately would be required on that page, i.e. `DB subnet group`

Reply:
@janboll, this is why we only pay attention to things that are sensible and safe to be allowed to postpone a Terraform apply run.

Everything that won't match against the list (of operations we allow to be deferred) will be a fall-through to the apply. Also, if you have pending changes i.e., **AttachedStorage** is included in the **PendingModifiedValues** property, if you decide to add something else alongside i.e., **EngineVersion**, then we will allow this to apply - to make sure that Terraform apply (provided that **apply_immediately** has not been set) can make the API call to ensure that this new change can be scheduled. However, if you then (there are two aforementioned pending changes) decide to apply changes immediately without waiting to the next maintenance window and set **apply_immediately**, then an apply will take place, and things will be applied immediately.

Does this make sense?

Reply:
Makes sense, perhaps rename it to `skippable_pending_ops`?

Reply:
Done. Thank you!